File: src/main/webapp/scripts/hello.js
(function () {

    function printMessage(message) {
        document.getElementById("message").innerHTML = message
    }

    var importObject = {
        env: {
            memory: new WebAssembly.Memory({ initial: 2, maximum: 10 }),
            printMessage
        }
    };

    WebAssembly.instantiateStreaming(fetch('wasm/hello.wasm'), importObject)
        .then(module => { module.instance.exports.hello() });

}());

File: src/main/java/ch/puzzle/wasm/hello/Hello.java
package ch.puzzle.wasm.hello;

import org.teavm.interop.Import;
import org.teavm.interop.Export;

public class Hello {

    @Import(module = "env", name = "printMessage")
    private static native void printMessage(int message);

    @Export(name = "hello")
    public static void hello() {
    	printMessage(42);
    }
}
